tests/ballerina-compiler-api-test/src/test/resources/test-src/typedesc_test.bal

```diff
@@ -216,3 +216,7 @@ type Pi PI;
 function testFunctionTypedesc() {
     function f = foo;
 }
+
+function testParameterizedType1(typedesc<anydata> td) returns td = external;
+
+function testParameterizedType2(typedesc td = <>) returns error|td = external;
```

Comment:
Shall we add a couple of cases where these get used as well? Basically to test whether the behaviour of the `type()` method in `SemanticModel` works as expected with dependently typed functions.

Reply:
Added in a3d9ef0.